src/server/stream_family.cc

```diff
@@ -1262,12 +1256,10 @@ OpResult<ClaimInfo> OpClaim(const OpArgs& op_args, string_view key, const ClaimO
       }
 
       // Try to get the consumer. If not found, create a new one.
-      op_args.shard->tmp_str1 =
-          sdscpylen(op_args.shard->tmp_str1, opts.consumer.data(), opts.consumer.size());
-      if ((consumer = streamLookupConsumer(cgr_res->cg, op_args.shard->tmp_str1, SLC_NO_REFRESH)) ==
-          nullptr) {
-        consumer = streamCreateConsumer(cgr_res->cg, op_args.shard->tmp_str1, nullptr, 0,
-                                        SCC_NO_NOTIFY | SCC_NO_DIRTIFY);
+      auto cname = WrapSds(opts.consumer);
+      if ((consumer = streamLookupConsumer(cgr_res->cg, cname, SLC_NO_REFRESH)) == nullptr) {
```

Comment:
nit: i personally do not like these double level expressions.

Reply:
Not me who introduced it ☝🏻